src/linear_algebra/special_linear_group.lean

```diff
@@ -60,25 +64,15 @@ namespace special_linear_group
 
 variables {n : Type u} [decidable_eq n] [fintype n] {R : Type v} [comm_ring R]
 
-instance coe_matrix : has_coe (special_linear_group n R) (matrix n n R) :=
+instance has_coe_to_matrix : has_coe (special_linear_group n R) (matrix n n R) :=
 ⟨λ A, A.val⟩
 
-instance coe_fun : has_coe_to_fun (special_linear_group n R) :=
-{ F   := λ _, n → n → R,
-  coe := λ A, A.val }
+local prefix `↑ₘ`:1024 := @coe _ (matrix n n R) _
```

Comment:
If I understand correctly, this is notation to fill in types that are constant throughout the file and hard for Lean to guess (i.e., for situations when Lean might know it needs a matrix but can't guess the types `n` and `R`).  And to be useful rather than unwieldy, the `n` and `R` are not explicit in the notation ... but this means it's only useful as `local` shorthand, not as global notation.

Is that right?  Could you add a comment along these lines, if so?

Reply:
```suggestion
/- In this file, Lean often has a hard time working out the values of `n` and `R` for an expression
like `det ↑A`. Rather than writing `(A : matrix n n R)` everywhere in this file which is annoyingly
verbose, or `A.val` which is not the simp-normal form for subtypes, we create a local notation
`↑ₘA`. This notation references the local `n` and `R` variables, so is not valid as a global
notation. -/
local prefix `↑ₘ`:1024 := @coe _ (matrix n n R) _
```